xpallets/gateway/bitcoin/v2/src/types.rs

```diff
@@ -0,0 +1,226 @@
+    pub(crate) to_be_redeemed_tokens: Balance,
+}
+
+#[derive(Encode, Decode, Default, Clone, PartialEq)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct Vault<AccountId, BlockNumber, Balance> {
+    /// Account identifier of the Vault
+    pub id: AccountId,
+    /// Number of tokens pending issue
+    pub to_be_issued_tokens: Balance,
+    /// Number of issued tokens
+    pub issued_tokens: Balance,
+    /// Number of tokens pending redeem
+    pub to_be_redeemed_tokens: Balance,
+    /// Bitcoin address of this Vault (P2PKH, P2SH, P2PKH, P2WSH)
+    pub wallet: BtcAddress,
+    /// Block height until which this Vault is banned from being
+    /// used for Issue, Redeem (except during automatic liquidation) and Replace .
+    pub banned_until: Option<BlockNumber>,
+    /// Current status of the vault
+    pub status: VaultStatus,
+}
+
+impl<AccountId: Default, BlockNumber: Default, Balance: Default>
+    Vault<AccountId, BlockNumber, Balance>
+{
+    pub(crate) fn new(id: AccountId, address: BtcAddress) -> Self {
+        Self {
+            id,
+            wallet: address,
+            ..Default::default()
+        }
+    }
+}
+
+#[derive(Encode, Decode, Clone, PartialEq)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub enum RedeemRequestStatus {
+    /// Redeem is accepted and vault will transfer btc
+    Processing,
+    /// Redeem is cancelled by redeemer
+    Cancelled,
+    /// Redeem is compeleted
+    Completed,
+}
+
+// Default value
+impl Default for RedeemRequestStatus {
+    fn default() -> Self {
+        RedeemRequestStatus::Processing
+    }
+}
+
+#[derive(Encode, Decode, Default, Clone, PartialEq)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct RedeemRequest<AccountId, BlockNumber, XBTC, PCX> {
+    /// Vault id
+    pub(crate) vault: AccountId,
+    /// Block height when the redeem requested
+    pub(crate) open_time: BlockNumber,
+    /// Who requests redeem
+    pub(crate) requester: AccountId,
+    /// Vault's btc address
+    pub(crate) btc_address: BtcAddress,
+    /// Amount that user wants to redeem
+    pub(crate) amount: XBTC,
+    /// Redeem fee amount
+    pub(crate) redeem_fee: PCX,
+    /// Request status
+    pub(crate) status: RedeemRequestStatus,
+    /// If redeem is reimbursed by redeemer
+    pub(crate) reimburse: bool,
+}
+
+/// Contains all informations while executing a issue request needed.
+#[derive(Encode, Decode, Default, Clone, PartialEq)]
+#[cfg_attr(feature = "std", derive(Debug, Deserialize, Serialize))]
+pub struct IssueRequest<AccountId, BlockNumber, XBTC, PCX> {
+    /// Vault id
+    pub(crate) vault: AccountId,
+    /// Block height when the issue requested
+    pub(crate) open_time: BlockNumber,
+    /// Who requests issue
+    pub(crate) requester: AccountId,
+    /// Vault's btc address
+    pub(crate) btc_address: BtcAddress,
+    /// Wheather request finished
+    pub(crate) completed: bool,
+    /// Wheather request cancelled
+    pub(crate) cancelled: bool,
```

Comment:
Why not use `IssueRequestStatus`? `RedeemRequestStatus` seemingly also applies here, we can just call it `RequestStatus` and use it for both `IssueRequest` and `RedeemRequest`.